### PR TITLE
refactor(card): clean question card

### PR DESCRIPTION
**Pull request by AGDholo**
## Related issues/discussions
#44 #26 

## Description of the changes

image

- create base card
- move question card to new `question folder`

- no style change
- broken change, need review

### CODE REVIEW

**Comment by izgzhen** [comment]
Taking a look now

File: src/components/Answer.vue [hunk 1]
```diff
@@ -319,9 +319,11 @@ import {
   commitSetWorkingDraft,
 } from '@/store/main/mutations';
 import { apiComment } from '@/api/comment';
+import BaseCard from '@/components/base/BaseCard.vue';
 
 @Component({
   components: {
+    BaseCard,
```
Comment (izgzhen) [inline src/components/Answer.vue:326]: Where is it used?

File: src/components/base/BaseCard.vue [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <v-sheet>
+    <v-card v-bind="$attrs" :class="{ 'c-card': !embedded }" :flat="embedded">
```
Comment (izgzhen) [inline src/components/base/BaseCard.vue:3]: I read that [v-sheet is the base for v-card](https://vuetifyjs.com/en/components/sheets/), so do we want use `v-card` here inside `v-sheet`? Is using `div` making more sense?

File: src/components/base/BaseCard.vue [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+@Component({
+  name: 'BaseCard',
+})
+export default class BaseCard extends Vue {
+  @Prop({ default: false }) private readonly embedded!: false;
```
Comment (izgzhen) [inline src/components/base/BaseCard.vue:18]: `: boolean`?

File: src/components/base/BaseCard.vue [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    <v-divider v-if="divider" class="mt-3" />
+  </v-sheet>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+@Component({
+  name: 'BaseCard',
```
Comment (izgzhen) [inline src/components/base/BaseCard.vue:15]: nit: `name` can be omitted if you don't use the component recursively.

**Review by izgzhen — CHANGES_REQUESTED** [review]

File: src/components/base/BaseCard.vue (hunk 2 above)
Comment (AGDholo) [reply src/components/base/BaseCard.vue:3]: This is an unnecessary label, it should delete all directly.

File: src/components/base/BaseCard.vue (hunk 3 above)
Comment (AGDholo) [reply src/components/base/BaseCard.vue:18]: yeah should boolean, I copy from other component

File: src/components/Answer.vue (hunk 1 above)
Comment (AGDholo) [reply src/components/Answer.vue:326]: should delete it, miss out it when i roll back the code

File: src/components/base/BaseCard.vue (hunk 2 above)
Comment (izgzhen) [reply src/components/base/BaseCard.vue:3]: > This is an unnecessary label, it should delete all directly.

What do you mean by "this", and "all"?

Comment (AGDholo) [reply src/components/base/BaseCard.vue:3]: `v-sheet` could extend theme provider and write some vuetify props, I suggest use `v-sheet` to replace `div` in most cases.

Comment (AGDholo) [reply src/components/base/BaseCard.vue:3]: https://vuetifyjs.com/en/components/sheets/

Comment (izgzhen) [reply src/components/base/BaseCard.vue:3]: Oh, my earlier comment is about the `v-card` inside the `v-sheet`. I was thinking whether the use of `v-card` is necessary.

Comment (AGDholo) [reply src/components/base/BaseCard.vue:3]: > Oh, my earlier comment is about the v-card inside the v-sheet. I was thinking whether the use of v-card is necessary.

Yes, but #45 should solve that repeated cards are used superimposed, I noticed a lot of unnecessary nesting of 'v-card'

File: src/components/base/BaseCard.vue [hunk 5]
```diff
@@ -9,14 +9,11 @@
 </template>
 
 <script lang="ts">
-import { Component, Prop, Vue } from 'vue-property-decorator';
+import { Prop, Vue } from 'vue-property-decorator';
 
-@Component({
```
Comment (izgzhen) [inline src/components/base/BaseCard.vue]: You still need the `@Component` decorator, just that `name` is not necessary.

Also, I recommend to show at least one use of `BaseCard` along with its definition.

**Review by izgzhen — CHANGES_REQUESTED** [review]

File: src/components/base/BaseCard.vue (hunk 5 above)
Comment (AGDholo) [reply src/components/base/BaseCard.vue]: LOL, I want to keep the name field,

How should I present a BaseCard example, I will use base-card for refactoring in my later submissions, but due to the length of the PR review, I only commit this component for now(Is that enough, or do I need to create a separate example file?)

**Comment by AGDholo** [comment]
Should I present a BaseCard example?

I want to use base-card for refactoring in my later commit, but due to the length of the PR review, I only commit this component for now(Is that enough, or do I need to create a separate example file?)

**Comment by izgzhen** [comment]
> Should I present a BaseCard example?

Right, like changing `QuestionPreview` or some another existing component to use it. So I can test it. No need for a separate test only file.

**Comment by izgzhen** [comment]
> > Should I present a BaseCard example?
> 
> Right, like changing `QuestionPreview` or some another existing component to use it. So I can test it. No need for a separate test only file.

Actually it is okay if you want to do it in a later commit. Both works for me.

**Comment by AGDholo** [comment]
> Right, like changing QuestionPreview or some another existing component to use it. So I can test it. No need for a separate test only file.

Gradually replacing it with ·base-card· could cause style inconsistencies

Padding:

image

**Review by izgzhen — APPROVED** [review]

**Comment by izgzhen** [comment]
> No-padding is truly style, but it needs more time to change all components

Sounds good!

**Pull request merged**